tracking/include/pcl/tracking/impl/tracking.hpp

```diff
@@ -462,6 +504,30 @@ struct EIGEN_ALIGN16 ParticleXYRPY : public _ParticleXYRPY {
         trans_x, 0, trans_z, trans_roll, trans_pitch, trans_yaw));
   }
 
+  template <class InputIterator>
+  static ParticleXYRPY
+  weightedAverage(InputIterator first, InputIterator last)
+  {
+    ParticleXYRPY wa;
+    float wa_roll_sin = 0.0, wa_roll_cos = 0.0, wa_pitch_sin = 0.0, wa_pitch_cos = 0.0,
+          wa_yaw_sin = 0.0, wa_yaw_cos = 0.0;
+    for (auto point = first; point != last; ++point) {
+      wa.x += point->x * point->weight;
+      wa.y += point->y * point->weight;
+      wa_pitch_cos = std::cos(point->pitch);
+      wa_roll_sin += wa_pitch_cos * std::sin(point->roll) * point->weight;
+      wa_roll_cos += wa_pitch_cos * std::cos(point->roll) * point->weight;
+      wa_pitch_sin += std::sin(point->pitch) * point->weight;
+      wa_yaw_sin += wa_pitch_cos * std::sin(point->yaw) * point->weight;
+      wa_yaw_cos += wa_pitch_cos * std::cos(point->yaw) * point->weight;
+    }
+    wa.z = 0;
```

Comment:
@KilianBerlo  Sorry, I just noticed: For `ParticleXYRPY`, `ParticleXYRP`, and `ParticleXYR`, the `y` component should be zero, not the `z` component. Please fix that, after that I am happy with the pull request.

Reply:
Thanks!